### PR TITLE
OP-1341 Enable Medical Type for Soft Deletion

### PR DESCRIPTION
**Pull request by mohit10696**
See OP-1341.

### CODE REVIEW

File: sql/step_04_all_following_steps.sql
Comment (mwithi) [inline sql/step_04_all_following_steps.sql:1]: please notice that in the while a new commit is landed on develop, so now "step_a113" is taken, you have to rename it to "step_a114", but better to do a merge with develop and resolve the conflicts

Comment (mohit10696) [reply sql/step_04_all_following_steps.sql:1]: Done

File: src/main/java/org/isf/medtype/service/MedicalTypeIoOperationRepository.java
Comment (dbmalkovsky) [inline src/main/java/org/isf/medtype/service/MedicalTypeIoOperationRepository.java]: This import is not used.

File: src/main/java/org/isf/medtype/service/MedicalTypeIoOperationRepository.java
Comment (dbmalkovsky) [inline src/main/java/org/isf/medtype/service/MedicalTypeIoOperationRepository.java]: This import is not used.

**Review by dbmalkovsky — COMMENTED** [review]
Why are there no associated unit tests to match this new code?

**Comment by mohit10696** [comment]
I was following this similar work (https://github.com/informatici/openhospital-core/pull/1336/ )for this and it has no test cases however I am implementing it....

File: src/main/java/org/isf/medtype/service/MedicalTypeIoOperationRepository.java
Comment (mohit10696) [reply src/main/java/org/isf/medtype/service/MedicalTypeIoOperationRepository.java]: Fixed

File: src/main/java/org/isf/medtype/service/MedicalTypeIoOperationRepository.java
Comment (mohit10696) [reply src/main/java/org/isf/medtype/service/MedicalTypeIoOperationRepository.java]: Modifying, Param is not used and removed, other are good

**Comment by mohit10696** [comment]
@dbmalkovsky  Test cases are here and removed all unused imports.

File: sql/load_demo_data.sql
Comment (mwithi) [inline sql/load_demo_data.sql]: Please always do a smoke test of the changes:

```suggestion
INSERT INTO `oh_medicaldsrtype` (`MDSRT_ID_A`, `MDSRT_DESC`, `MDSRT_DELETED`, `MDSRT_CREATED_BY`, `MDSRT_CREATED_DATE`, `MDSRT_LAST_MODIFIED_BY`, `MDSRT_LAST_MODIFIED_DATE`, `MDSRT_ACTIVE`) VALUES ('D','Drugs','N',NULL,NULL,NULL,NULL,1);
INSERT INTO `oh_medicaldsrtype` (`MDSRT_ID_A`, `MDSRT_DESC`, `MDSRT_DELETED`, `MDSRT_CREATED_BY`, `MDSRT_CREATED_DATE`, `MDSRT_LAST_MODIFIED_BY`, `MDSRT_LAST_MODIFIED_DATE`, `MDSRT_ACTIVE`) VALUES ('K','Chemical','N',NULL,NULL,NULL,NULL,1);
INSERT INTO `oh_medicaldsrtype` (`MDSRT_ID_A`, `MDSRT_DESC`, `MDSRT_DELETED`, `MDSRT_CREATED_BY`, `MDSRT_CREATED_DATE`, `MDSRT_LAST_MODIFIED_BY`, `MDSRT_LAST_MODIFIED_DATE`, `MDSRT_ACTIVE`) VALUES ('L','Laboratory','N',NULL,NULL,NULL,NULL,1);
INSERT INTO `oh_medicaldsrtype` (`MDSRT_ID_A`, `MDSRT_DESC`, `MDSRT_DELETED`, `MDSRT_CREATED_BY`, `MDSRT_CREATED_DATE`, `MDSRT_LAST_MODIFIED_BY`, `MDSRT_LAST_MODIFIED_DATE`, `MDSRT_ACTIVE`) VALUES ('S','Surgery','N',NULL,NULL,NULL,NULL,1);
```

**Review by mwithi — CHANGES_REQUESTED** [review]
Just a little fix on demo data, after that LGTM.

**Review by dbmalkovsky — DISMISSED** [review]
I agree with @mwithi comments; LGTM after the change

**Comment by mohit10696** [comment]
Hi @mwithi , Done with the changes. Not sure what went wrong. Merging is still blocked by requested changes.

**Review by mwithi — APPROVED** [review]

**Comment by mwithi** [comment]
> Hi @mwithi , Done with the changes. Not sure what went wrong. Merging is still blocked by requested changes.

Because I'm the one who has to accept them. Now is ok, let's finish the checks and then we can merge. Thanks!

**Comment by mohit10696** [comment]
> > Hi @mwithi , Done with the changes. Not sure what went wrong. Merging is still blocked by requested changes.
> 
> Because I'm the one who has to accept them. Now is ok, let's finish the checks and then we can merge. Thanks!

You can merge now

**Pull request merged**
